ern/index.go

```diff
@@ -268,55 +273,165 @@ func (i *Indexer) indexResourceList(ernID *cid.Cid, obj *meta.Object) error {
-		title = v.(string)
+		title = rt.(string)
 	} else if !meta.IsPathNotFound(err) {
 		return err
 	}
 
-	// return an error if there is neither an ID nor a ReferenceTitle
-	if len(ids) == 0 && title == "" {
-		return fmt.Errorf("SoundRecording missing both SoundRecordingId and ReferenceTitle")
+	// return an error if there is no ReferenceTitle, SoundRecordingId can be empty
+	if title == "" {
+		return fmt.Errorf("SoundRecording missing ReferenceTitle")
 	}
 
-	// update the sound_recording and resource_list indexes with each ID
-	for _, id := range ids {
-		_, err := i.db.Exec(
-			"INSERT INTO sound_recording (cid, id, title) VALUES ($1, $2, $3)",
-			obj.Cid().String(), id, title,
-		)
-		if err != nil {
-			return err
+	// update the sound_recording and resource_list indexes
+	if i.db.Exec(
+		"INSERT INTO sound_recording (cid, id, title) VALUES ($1, $2, $3)",
+		obj.Cid().String(), isrc, title,
+	); err != nil {
+		return err
+	}
+
+	if i.db.Exec(
+		"INSERT INTO resource_list (ern_id, resource_id) VALUES ($1, $2)",
+		ernID.String(), obj.Cid().String(),
+	); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// indexReleaseList indexes the ReleaseList for each Release composite
+func (i *Indexer) indexReleaseList(ernID *cid.Cid, metaObj *meta.Object) error {
+	// Much like the resource list, the release propoerty can be
+	// a single release, or an array of links.
+	rls, err := metaObj.Get("Release") 
+	if err != nil {
+		return err
+	}
+	var cids []*cid.Cid
+	switch rls := rls.(type) {
+	case *format.Link:
+		cids = []*cid.Cid{rls.Cid}
+	case []interface{}:
+		for _, x := range rls {
+			id, ok := x.(*cid.Cid)
+			if !ok {
+				return fmt.Errorf("Invalid release type %T, expected *cid.Cid", x)
+			}
+			cids = append(cids, id)
 		}
+	}
 
-		_, err = i.db.Exec(
-			"INSERT INTO resource_list (ern_id, resource_id) VALUES ($1, $2)",
-			ernID.String(), obj.Cid().String(),
-		)
+	// load and index each Release link
+	for _, id := range cids {
+		rObj, err := i.store.Get(id)
 		if err != nil {
 			return err
 		}
+		if err := i.indexRelease(ernID, rObj); err != nil {
+			return err
+		}
 	}
-
 	return nil
 }
 
-func (i *Indexer) indexReleaseList(ernID *cid.Cid, obj *meta.Object) error {
-	// TODO: index Releases
+// indexRelease index each Release composite in the ReelaseList
+func (i * Indexer) indexRelease(ernID *cid.Cid, metaObj *meta.Object) error {
+
+	graph := meta.NewGraph(i.store, metaObj)
+
+	// Only *attempt* to load the GRid, other IDs can be retrieved via GraphQL
+	var grId string
+	v, err := graph.Get("ReleaseId", "GRid", "@value")
+	if err == nil {
+		grId = v.(string)
+	}
+
+	// load the ReferenceTitle
+	var title string
+	rtl, err := graph.Get("ReferenceTitle", "TitleText", "@value")
+	if err == nil {
+		title = rtl.(string)
+	} else if !meta.IsPathNotFound(err) {
+		return err
+	}
+
+	// Index ReleaseType
+	// SQLite index threw - revisit
+	// If the field is not present, default to empty string
+	// var rType string
```

Comment:
Remove this comment

Reply:
Sure, don't mind doing so, it is intended as a TODO, as it would be beneficial to have a index for the types of releases available.